tsconfig.json

```diff
@@ -19,5 +19,10 @@
     "allowSyntheticDefaultImports": true,
     "skipLibCheck": true
   },
-  "exclude": ["examples", "es"]
+  "exclude": [
+    "examples",
+    "es",
+    "src/middlewares/__tests__/createInsightsMiddleware.ts",
+    "test/mock/createInsightsClient.ts"
```

Comment:
This test file imports `createInsightsClient` from `test/mock/createInsightsClient.ts` and it imports it from `node_modules/search-insights`.

Unless we exclude these two files, `tsc` will type-check `search-insights` and fail.
I tried to avoid it without having to exclude the test file here, but

- https://github.com/Microsoft/TypeScript/issues/14293
- https://github.com/microsoft/TypeScript/issues/36017
- https://www.reddit.com/r/typescript/comments/747xz7/suppress_errors_in_node_modules/

> "exclude" only prevents items from being included by "include"; it doesn't prevent them from being included via import statements or <reference>s. If you import levelup, and levelup imports leveldown, then excluding leveldown or node_modules won't have an effect.

couldn't find a way.

Reply:
is skipLibCheck really not an option?

Reply:
or fixing the TS of search-insights not to have those issues?

Reply:
> skipLibCheck

It's already turned on in our tsconfig but it doesn't affect here. (tried to pass it via CLI too)

> or fixing the TS of search-insights not to have those issues?

We could try. But, running `tsc` in `search-insights` already tells me everything's fine probably because we have different tsconfig and different version of tsc.

The possible next step would be to figure out the way to modify either the source or tsconfig of search-insights so that it works fine in InstantSearch.js without having to exclude these.

Reply:
To be clear, what I meant by "The possible next step" was in a separate PR. I'd like to get this PR merged with no great amount of further investment at the moment. We can revisit and do the "next step" later. cc @Haroenv @tkrugg

Reply:
unfortunately I don't think we can merge a PR with so many failing tests

Reply:
Oh, I didn't notice that. I'll take a look.